src/core/symbology-ng/qgssymbol.cpp

```diff
@@ -842,30 +843,49 @@ void QgsSymbol::renderFeature( const QgsFeature& feature, QgsRenderContext& cont
       const QgsGeometryCollection& geomCollection = dynamic_cast<const QgsGeometryCollection&>( *segmentizedGeometry.geometry() );
       const unsigned int num = geomCollection.numGeometries();
 
+      // Sort components by approximate area (probably a bit faster than using
+      // area() )
+      std::map<double, QList<unsigned int> > mapAreaToPartNum;
```

Comment:
I needed a reverse iterator and couldn't find one in QMap API, unless I missed it